core/arch/arm/kernel/thread_spmc.c

```diff
@@ -393,24 +367,45 @@ static void handle_partition_info_get(struct thread_smc_args *args,
 	spmc_set_args(args, ret_fid, FFA_PARAM_MBZ, rc, FFA_PARAM_MBZ,
 		      FFA_PARAM_MBZ, FFA_PARAM_MBZ);
 }
+#endif /*CFG_CORE_SEL1_SPMC*/
 
 static void handle_yielding_call(struct thread_smc_args *args)
 {
-	uint32_t ret_val = 0;
+	TEE_Result res = 0;
 
 	thread_check_canaries();
 
 	if (args->a3 == OPTEE_FFA_YIELDING_CALL_RESUME) {
 		/* Note connection to struct thread_rpc_arg::ret */
 		thread_resume_from_rpc(args->a7, args->a4, args->a5, args->a6,
 				       0);
-		ret_val = FFA_INVALID_PARAMETERS;
+		res = TEE_ERROR_BAD_PARAMETERS;
 	} else {
-		thread_alloc_and_run(args->a1, args->a3, args->a4, args->a5);
-		ret_val = FFA_BUSY;
+		thread_alloc_and_run(args->a1, args->a3, args->a4, args->a5,
```

Comment:
@jenswi-linaro I just noticed that args->a6 and args->a7 won't be used in thread_alloc_and_run is this expected?

Reply:
It does look odd, `args->a6` is even supposed to be used.

Reply:
Fix in https://github.com/OP-TEE/optee_os/pull/4609